### PR TITLE
LA workspace workflow and associated refactoring

### PR DESCRIPTION
**Pull request by VaclavK**

### CODE REVIEW

**Review by rmjoia — APPROVED** [review]

File: src/Mammon/Actors/IResourceActor.cs [hunk 1]
```diff
@@ -13,8 +13,17 @@ public interface IResourceActor : IActor
...
+    Task<(string costCentre, ResourceCost cost)> AssignCostCentre();
+
+    /// <summary>
+    /// returns the assigned cost centre for this resource and flag indicating if it was assigned
+    /// 
+    /// actor NOT assigned cost centre would indicate that the particular actor id has not been seen yet
+    /// (e.g. perhaps not bearing a direct cost and thus not returned by Cost API)s
+    /// </summary>
+    /// <returns>tuple with cost centre and assigned flag</returns>
+    Task<(string? costCentre, bool assigned)> GetAssignedCostCentre();
```
[the remark this answers is not in the document]
Comment (VaclavK) [reply src/Mammon/Actors/IResourceActor.cs:28]: I have done that as I have "setter" for that - AssignCostCentre  - so I have getter...

File: src/Mammon/Services/CostCentreRuleEngine.cs [hunk 2]
```diff
@@ -7,129 +7,140 @@
...
+	private readonly IConfiguration configuration;
+	private IList<CostCentreRule> CostCentreRules { get; set; } = [];
+	public IList<SubscriptionDefinition> Subscriptions { get; internal set; } = [];
+	public IEnumerable<string> CostCentres { get; internal set; } = [];
+	public string DefaultCostCentre { get; set; } = string.Empty;
+	public IEnumerable<string> ResourceGroupSuffixRemoveList { get; internal set; } = [];
+	public IDictionary<string, string> ResourceGroupTokenClassMap { get; internal set; } = new Dictionary<string, string>();
+	public IEnumerable<string> SubscriptionNames { get; internal set; } = [];
 	public IList<string> SpecialModes { get; set; } = [];
+	public IDictionary<string, string> AKSNamespaceMapping { get; set; } = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
```
[the remark this answers is not in the document]
Comment (VaclavK) [reply src/Mammon/Services/CostCentreRuleEngine.cs:19]: good point
while this is a new feature in the cost centre definition file - waiting for the mapping from David - I can already connect it in the code - overlooked that

**Comment by VaclavK** [comment]
@artpatskou  PR feedback addressed, please check

File: src/Mammon/Actors/IResourceActor.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (VaclavK) [reply src/Mammon/Actors/IResourceActor.cs]: would assignmentExists work?

**Pull request merged**